specs/_features/eip7732/beacon-chain.md

```diff
@@ -322,6 +373,47 @@ def remove_flag(flags: ParticipationFlags, flag_index: int) -> ParticipationFlag
 
 ### Predicates
 
+#### New `is_attestation_same_slot`
+
+```python
+def is_attestation_same_slot(state: BeaconState, data: AttestationData) -> bool:
+    """
+    Checks if the attestation was for the block proposed at the attestation slot
+    """
+    if data.slot == 0:
+        return True
+    is_matching_blockroot = data.beacon_block_root == get_block_root_at_slot(state, Slot(data.slot))
+    is_current_blockroot = data.beacon_block_root != get_block_root_at_slot(
+        state, Slot(data.slot - 1)
+    )
+    return is_matching_blockroot and is_current_blockroot
+```
+
+#### Modified `is_compounding_withdrawal_credential`
+
+```python
+def is_compounding_withdrawal_credential(withdrawal_credentials: Bytes32) -> bool:
+    prefix = withdrawal_credentials[:1]
+    return prefix == COMPOUNDING_WITHDRAWAL_PREFIX or prefix == BUILDER_WITHDRAWAL_PREFIX
+```
```

Comment:
This is a very misleading change. We should update this function instead:
* [`has_execution_withdrawal_credential`](https://github.com/ethereum/consensus-specs/blob/master/specs/electra/beacon-chain.md#new-has_execution_withdrawal_credential)

And rename `is_builder` to `has_builder_withdrawal_credential` for consistency.

Reply:
Oops, I meant to link `has_compounding_withdrawal_credential` not the execution one.